tests/src/test/java/edu/kit/kastel/mcse/ardoco/core/tests/eval/EvaluationMetrics.java

```diff
@@ -150,6 +154,51 @@ public static double calculatePhiCoefficient(int truePositives, int falsePositiv
         return num.divide(denominator, MathContext.DECIMAL128).doubleValue();
     }
 
+    /**
+     * Calculates the maximum possible value of the phi coefficient given the four values of the confusion matrix (TP, FP, FN, TN).
+     *
+     * @see <a href="https://link.springer.com/article/10.1007/BF02288588">Paper about PhiMax by Ferguson (1941)</a>
+     * @see <a href="https://journals.sagepub.com/doi/abs/10.1177/001316449105100403">Paper about Phi/PhiMax by Davenport et al. (1991)<</a>
+     * @param truePositives  number of true positives
+     * @param falsePositives number of false positives
+     * @param falseNegatives number of false negatives
+     * @param trueNegatives  number of true negatives
+     * @return The maximum possible value of phi.
+     */
+    public static double calculatePhiCoefficientMax(int truePositives, int falsePositives, int falseNegatives, int trueNegatives) {
+        var tp = BigDecimal.valueOf(truePositives);
+        var fp = BigDecimal.valueOf(falsePositives);
+        var fn = BigDecimal.valueOf(falseNegatives);
+        var tn = BigDecimal.valueOf(trueNegatives);
+
+        var test = fn.add(tp).compareTo(fp.add(tp)) >= 0;
```

Comment:
@Gram21 it that test really correct ?

`fn.add(tp).compareTo(fp.add(tp)) >= 0;` should be equivalent to `fn+tp >= fp+tp` , right?

Reply:
Yes, it should be equivalent to `fn+tp >= tp+fp`, as written in the Davenport paper. There, it states "Given P2 >= P1" and you can translate it, when assuming `p11 = tp`, to `P2=fn+tp` and `P1=fp+tp`